File: src/explore-education-statistics-common/src/services/themeService.ts
import { contentApi } from '@common/services/api';
import { MethodologySummary } from '@common/services/types/methodology';

export interface PublicationMethodologySummary {
  id: string;
  title: string;
  methodologies: MethodologySummary[];
}
export interface MethodologyTheme {
  id: string;
  title: string;
  summary: string;
  publications: PublicationMethodologySummary[];
}

export interface ThemeSummary {
  id: string;
  slug: string;
  summary: string;
  title: string;
}

const themeService = {
  getMethodologyThemes(): Promise<MethodologyTheme[]> {
    return contentApi.get(`/methodology-themes`);
  },
  listThemes(): Promise<ThemeSummary[]> {
    return contentApi.get(`/themes`);
  },
};

export default themeService;

File: src/GovUk.Education.ExploreEducationStatistics.Content.Services/ThemeService.cs
using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
using GovUk.Education.ExploreEducationStatistics.Content.Services.Interfaces;
using GovUk.Education.ExploreEducationStatistics.Content.ViewModels;
using Microsoft.EntityFrameworkCore;

namespace GovUk.Education.ExploreEducationStatistics.Content.Services;

public class ThemeService(ContentDbContext contentDbContext) : IThemeService
{
    public async Task<IList<ThemeViewModel>> ListThemes() =>
        await contentDbContext.Themes
            .Where(theme =>
                theme.Publications.Any(publication =>
                    publication.LatestPublishedReleaseVersionId.HasValue &&
                    (publication.SupersededById == null ||
                     !publication.SupersededBy!.LatestPublishedReleaseVersionId.HasValue)))
            .OrderBy(theme => theme.Title)
            .Select(theme => new ThemeViewModel
            {
                Id = theme.Id,
                Slug = theme.Slug,
                Title = theme.Title,
                Summary = theme.Summary
            }).ToListAsync();
}
